From 3632b4da11866056dc8a003b9a9d97b726a8aca2 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Fri, 10 Mar 2017 18:33:40 +0200
Subject: [PATCH 1560/5242] MLK-14425: clk-imx6sx: Check clk_on_imx6sx in
 imx_amp_power_init

commit  ceaa95f74fa781359f00403a6709b835d020b986 from
https://source.codeaurora.org/external/imx/linux-imx.git

This fixes crashing on boot on imx7d if imx_src_is_m4_enabled(). This
check was lost when porting 24ee04ce76d6a8d8229c642eb5d7a15ea57570c6.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx6sx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/imx/clk-imx6sx.c b/drivers/clk/imx/clk-imx6sx.c
index 7f9900f..4ee25ad 100644
--- a/drivers/clk/imx/clk-imx6sx.c
+++ b/drivers/clk/imx/clk-imx6sx.c
@@ -697,7 +697,7 @@ static int __init imx_amp_power_init(void)
 	int i;
 	void __iomem *shared_mem_base;
 
-	if (!(imx_src_is_m4_enabled()))
+	if (!(imx_src_is_m4_enabled() && clk_on_imx6sx()))
 		return 0;
 
 	amp_power_mutex = imx_sema4_mutex_create(0, MCC_POWER_SHMEM_NUMBER);
-- 
1.7.9.5

